Cute/Impl/FileBasedPersister.swift

```
@@ -101,15 +109,23 @@ public class FileBasedPersister<HandlingJob: QueueJob>: JobPersister {
     ///
     /// - Parameter completion: The block to call after the attempt to clear all jobs is complete.
     public func clear(completion: ((Error?) -> Void)?) {
+
+        let files: [URL]
         do {
-            let path = fileManager.currentDirectoryPath
-            try fileManager.removeItem(atPath: path)
-            try fileManager.createDirectory(atPath: path, withIntermediateDirectories: true)
-            fileManager.changeCurrentDirectoryPath(path)
-        } catch let error {
-            completion?(error)
+            files = try fileManager.contentsOfDirectory(at: persistenceLocation, includingPropertiesForKeys: [], options: .skipsSubdirectoryDescendants)
+        } catch {
+            // a directory that cannot be read is almost the same as an empty directory, right? I think it's safe to ignore
+            files = []
```

I've gone back and forth on this. At this level (Framework), I think it's best to bubble up the error and let the consumer determine what they want to do with this. That error could point to a bug in their environment, setup, etc. Given that we don't know what the error is, or that we care at this level (and that Cute actually supports tvOS and macOS), I think it's best to let the error bubble up.

---

Yes, makes sense. Almost all of the methods in FilePersister throw, so might as well be consistent. I did run into one problem... in the drain method i cannot throw from within a dispatched block. So, i just had to print the error out like an animal.